src/components/game/components/board/Board.js

```diff
@@ -1,198 +1,70 @@
-      return;
-    }
-
-    return <Replace onClick={onClickHandler} />;
+  const renderReplaceCardButton = () => {
+    if (phase <= 2 || phase >= 4) {
+      return null;
+    } else return <Replace onClick={onClickReplaceHandler} />;
   };
 
   const renderFoldButton = () => {
-    const onClickHandler = () => {
-      const newPlayerMoney = state.playerMoney;
-      const newAiMoney = state.aiMoney;
-      const newPlayerBet = state.playerBet;
-      const newAiBet = state.aiBet;
-
-      if (state.playerHand.length === 5) {
-        setState(() => ({
-          ...initialState,
-          playerMoney: newPlayerMoney - newPlayerBet,
-          aiMoney: newAiMoney + (newAiBet + newPlayerBet),
-        }));
-
-        alert("You lose!");
-      }
-    };
-
-    if (state.phase !== 2 && state.phase < 4) {
-      return;
-    }
-
-    return <Fold onClick={onClickHandler} />;
+    if (phase !== 2 && phase < 4) {
+      return null;
+    } else return <Fold onClick={onClickFoldHandler} />;
   };
 
   const renderCheckButton = () => {
-    const onClickHandler = () => {
-      const newPlayerMoney = state.playerMoney;
-      const newAiMoney = state.aiMoney;
-      const newPlayerBet = state.playerBet;
-      const newAiBet = state.aiBet;
-
-      if (state.playerHand.length === 5) {
-        if (handCheck(state.playerHand) > handCheck(state.aiHand)) {
-          setState(() => ({
-            ...initialState,
-            playerMoney: newPlayerMoney + (newPlayerBet + newAiBet),
-            aiMoney: newAiMoney - newAiBet,
-          }));
-
-          alert("You win!");
-        } else if (handCheck(state.playerHand) === handCheck(state.aiHand)) {
-          setState(() => ({
-            ...initialState,
-            playerMoney: newPlayerMoney + newPlayerBet,
-            aiMoney: newAiMoney + newAiBet,
-          }));
-
-          alert("Tie!");
-        } else {
-          setState(() => ({
-            ...initialState,
-            playerMoney: newPlayerMoney - newPlayerBet,
-            aiMoney: newAiMoney + (newPlayerBet + newAiBet),
-          }));
-
-          alert("You lose!");
-        }
-      }
-    };
-
-    if (state.phase !== 2 && state.phase < 4) {
-      return;
-    }
-
-    return <Check onClick={onClickHandler} />;
+    if (phase !== 2 && phase < 4) {
+      return null;
+    } else return <Check onClick={onClickCheckHandler} />;
   };
 
   const renderRaiseButton = () => {
-    const onClickHandler = () => {
-      const newPlayerBet = state.playerBet;
-      const newAiBet = state.aiBet;
-      const bet = 100;
-
-      setState((prevState) => ({
-        ...prevState,
-        playerBet: newPlayerBet + bet,
-        aiBet: newAiBet + bet,
-      }));
-    };
-
-    if (
-      state.playerMoney === state.playerBet ||
-      (state.phase !== 2 && state.phase < 4)
-    ) {
-      return;
-    }
-
-    return <Raise onClick={onClickHandler} />;
+    if (playerMoney === playerBet || (phase !== 2 && phase < 4)) {
```

Comment:
This comment applies for more occurrences.
Rendering of an element should occur by the decision of the container element. For example the parent component should have the if that will allow the render of a component.

By doing this we achieve purity (avoid using a prop that's not a part of our scope (phase in this case).
It is reusable (you can make button component that will have props and render differently based on its props

Reply:
Done.